common/lib/dependabot/dependency_updaters/README.md

```diff
@@ -0,0 +1,72 @@
+# Dependency Updater
+
+A *dependency updater* is responsible for updating a dependent package
+according to a set of requirements.
+It provides a minimal interface for languages to add Dependabot support
+by delegating responsibilities to a package manager for that language.
+
+## Public API
+
+Each `Dependabot::DependencyUpdaters` class inherits from `Base` 
+and implements the following methods:
+
+### `#update(dependency:, requirements:)`
+
+Updates a dependency according to the provided requirements.
+This method is called within the root directory of a project.
```

Comment:
We currently support setting the `directory` so running commands in some sub-project, do we want to support this?

Reply:
Yes, I think we should, so maybe:

```suggestion
This method is called within the directory specified in `configuration.directory` or the root directory of a project if `configuration.directory` is `nil`.
```

Or alternatively, make the default `configuration.directory` be the root, and ensure it's never nil because nil sucks and then we can say:

```suggestion
This method is called within the directory specified in `configuration.directory`, which is the root directory of the project by default.
```